# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoEncoder.cpp
# Commit: ea4d0db67f89
# Full Hash: ea4d0db67f899714106e6f4ffcc78ebe8feaf466
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Implement plumbing for SVC. r=chunmin,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D196473
# ==============================================================================

diff -r f0ac27321cf8 -r ea4d0db67f89 dom/media/webcodecs/VideoEncoder.cpp
--- a/dom/media/webcodecs/VideoEncoder.cpp	Fri Dec 22 21:39:18 2023 +0000
+++ b/dom/media/webcodecs/VideoEncoder.cpp	Fri Dec 22 21:39:18 2023 +0000
@@ -250,13 +250,25 @@
 #endif
     specific.emplace(VP9Specific());
   }
+  MediaDataEncoder::ScalabilityMode scalabilityMode;
+  if (mScalabilityMode) {
+    if (mScalabilityMode->EqualsLiteral("L1T2")) {
+      scalabilityMode = MediaDataEncoder::ScalabilityMode::L1T2;
+    } else if (mScalabilityMode->EqualsLiteral("L1T3")) {
+      scalabilityMode = MediaDataEncoder::ScalabilityMode::L1T3;
+    } else {
+      scalabilityMode = MediaDataEncoder::ScalabilityMode::None;
+    }
+  } else {
+    scalabilityMode = MediaDataEncoder::ScalabilityMode::None;
+  }
   return EncoderConfig(
       codecType, {mWidth, mHeight}, usage, ImageBitmapFormat::RGBA32, ImageBitmapFormat::RGBA32,
       AssertedCast<uint8_t>(mFramerate.refOr(0.f)), 0, mBitrate.refOr(0),
       mBitrateMode == VideoEncoderBitrateMode::Constant
           ? MediaDataEncoder::BitrateMode::Constant
           : MediaDataEncoder::BitrateMode::Variable,
-      hwPref, specific);
+      hwPref, scalabilityMode, specific);
 }
 already_AddRefed<WebCodecsConfigurationChangeList>
 VideoEncoderConfigInternal::Diff(
@@ -343,8 +355,10 @@
 
   // Not supported
   if (aConfig->mScalabilityMode.isSome() &&
-      !aConfig->mScalabilityMode->IsEmpty()) {
-    LOGE("Scalability mode not supported");
+      !aConfig->mScalabilityMode->EqualsLiteral("L1T2") &&
+      !aConfig->mScalabilityMode->EqualsLiteral("L1T3")) {
+    LOGE("Scalability mode %s not supported",
+         NS_ConvertUTF16toUTF8(aConfig->mScalabilityMode.value()).get());
     return false;
   }
 